packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/AdminsAndModerators.tsx

```diff
@@ -1,7 +1,174 @@
+    searchTerm: debouncedSearchTerm,
+    limit: 20,
+    orderBy: APIOrderBy.LastActive,
+    orderDirection: APIOrderDirection.Desc,
+    includeRoles: true,
+  });
+
+  const roleData = useMemo(() => {
+    if (!searchResults?.pages?.length) {
+      return [];
+    }
+    return searchResults.pages[0].results.map((profile) => {
+      return {
+        ...(profile.roles[0] || {}),
+        Address: profile.addresses[0],
+        id: profile.addresses[0].id,
+        displayName: profile.profile_name || 'Anonymous',
+      };
+    });
+  }, [searchResults]);
+
+  const handleRoleUpdate = (oldRole, newRole) => {
+    // newRole doesn't have the Address property that oldRole has,
+    // Add the missing Address property to the newRole, then splice it into the array.
+    newRole.Address = oldRole.Address;
+
+    const predicate = (r) => {
+      return r.address_id === oldRole.address_id;
+    };
+
+    app.roles.addRole(newRole);
+    app.roles.removeRole(predicate);
+
+    const { adminsAndMods } = app.chain.meta;
+
+    if (oldRole.permission === 'admin' || oldRole.permission === 'moderator') {
+      const idx = adminsAndMods.findIndex(predicate);
+
+      if (idx !== -1) {
+        adminsAndMods.splice(idx, 1);
+      }
+      if (oldRole.permission === 'admin') {
+        setAdmins(admins.filter((a) => a.address_id !== oldRole.address_id));
+      }
+      if (oldRole.permission === 'moderator') {
+        setMods(mods.filter((a) => a.address_id !== oldRole.address_id));
+      }
+    }
+
+    if (newRole.permission === 'admin' || newRole.permission === 'moderator') {
+      const roleInfo = new RoleInfo(
+        newRole.address_id,
+        newRole.Address?.id || newRole.address_id,
+        newRole.Address.address,
+        newRole.Address.community_id,
+        newRole.chain_id,
+        newRole.permission,
+        newRole.allow,
+        newRole.deny,
+        newRole.is_user_default,
+      );
+      adminsAndMods.push(roleInfo);
+
+      if (newRole.permission === 'admin') {
+        setAdmins([...admins, newRole]);
+      }
+      if (newRole.permission === 'moderator') {
+        setMods([...mods, newRole]);
+      }
+    }
+
+    refetch();
+  };
+
+  return (
+    <section className="AdminsAndModerators">
+      <section className="left-section">
+        <div className="header">
+          <CWText type="h2">Admins and Moderators</CWText>
+          <CWText type="b1">
+            Let&apos;s start with some basic information about your community
+          </CWText>
+        </div>
+        <ManageRoles
+          label="Admins"
+          roledata={admins}
+          onRoleUpdate={handleRoleUpdate}
+        />
+        <CWText type="caption" className={ComponentType.Label}>
+          Members
+        </CWText>
+
+        <UpgradeRolesForm
+          roleData={roleData}
+          onRoleUpdate={handleRoleUpdate}
+          searchTerm={searchTerm}
+          setSearchTerm={setSearchTerm}
+        />
```

Comment:
We should change the checkbox and buttons, and use new designs 
<img width="564" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/51641047/8dda6705-c09d-4f35-8d2b-5c72e10d9cf4">